esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchAdapter.java

```diff
@@ -91,21 +94,45 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpCli
         request.scroll(scroll);
         SearchResponse response = null;
         TabularQueryResult results = null;
-        List<Map<String, Object>> scrolledResults = null;
+        List<Map<String, Object>> collectedResults = null;
 
         try {
-            response = this.client.search(request, RequestOptions.DEFAULT);
+            if (query.getLimitClause() != null && query.getLimitClause().getLimit() > ES_BATCH_LIMIT
+                    && query.getAggregateClauses() != null && !query.getAggregateClauses().isEmpty()) {
+                int numPartitions = 10;
+
+                TermsAggregationBuilder termsAB = null;
+                        Collection<AggregationBuilder> aggregationBuilders = request.source().aggregations()
+                                .getAggregatorFactories();
+                for (AggregationBuilder aggregation : aggregationBuilders) {
+                    if (aggregation instanceof TermsAggregationBuilder) {
+                        termsAB = (TermsAggregationBuilder) aggregation;
+                    }
+                }
+                if (termsAB != null) {
+                    collectedResults = new ArrayList<>();
+                    for (int i = 0; i < numPartitions; i++) {
+                        termsAB.includeExclude(new IncludeExclude(i, numPartitions));
+                        SearchResponse partitionResponse = this.client.search(request, RequestOptions.DEFAULT);
+                        collectedResults.addAll(ElasticsearchResultsConverter.convertResults(query, partitionResponse));
+                    }
+                    collectedResults = ElasticsearchResultsConverter.sortBuckets(query.getOrderByClauses(), collectedResults);
+                }
+            } else {
+                response = this.client.search(request, RequestOptions.DEFAULT);
+            }
+
             if (query.getLimitClause() != null && query.getLimitClause().getLimit() > ES_BATCH_LIMIT) {
-                scrolledResults = ElasticsearchResultsConverter.getScrolledResults(scroll, response, this.client);
+                collectedResults = ElasticsearchResultsConverter.getScrolledResults(scroll, response, this.client);
             }
```

Comment:
@jacklechnercaci So if I change the Limit on the Networks Bar Chart (Earthquakes dataset) to `40000` I get a NullPointerException on line 126.  I think this whole "if" statement should be inside the "else" statement on line 121, since it's not relevant to aggregation queries?

Reply:
Yep. Changed it to an else if under the aggregation if.

Reply:
Awesome.